webapi/src/main/scala/org/knora/webapi/messages/util/search/QueryTraverser.scala

```diff
@@ -183,20 +202,117 @@ trait ConstructToSelectTransformer extends WhereTransformer {
  * or [[ConstructToSelectTransformer]].
  */
 object QueryTraverser {
+  private implicit val stringFormatter: StringFormatter = StringFormatter.getGeneralInstance
+  private implicit val timeout: Timeout                 = Duration(5, SECONDS)
+
+  /**
+   * Helper method that analyzed an RDF Entity and returns a sequence of Ontology IRIs that are being referenced by the entity.
+   * If an IRI appears that can not be resolved by the ontology cache, it will check if the IRI points to project data;
+   * if so, all ontologies defined by the project to which the data belongs, will be included in the results.
+   *
+   * @param entity       an RDF entity.
+   * @param map          a map of entity IRIs to the IRIs of the ontology where they are defined.
+   * @param storeManager a reference to the storeManager to retrieve a [[ProjectADM]] by a shortcode.
+   * @return a sequence of ontology IRIs which relate to the input RDF entity.
+   */
+  private def resolveEntity(entity: Entity, map: Map[SmartIri, SmartIri], storeManager: ActorRef): Seq[SmartIri] =
+    entity match {
+      case IriRef(iri, _) => {
+        val internal     = iri.toOntologySchema(InternalSchema)
+        val maybeOntoIri = map.get(internal)
+        maybeOntoIri match {
+          // if the map contains an ontology IRI corresponding to the entity IRI, then this can be returned
+          case Some(iri) => Seq(iri)
+          case None => {
+            // if the map doesn't contain a corresponding ontology IRI, then the entity IRI points to a resource or value
+            // in that case, all ontologies of the project, to which the entity belongs, should be returned.
+            val shortcode = internal.getProjectCode
+            shortcode match {
+              case None => Seq.empty
+              case some => {
```

Comment:
```suggestion
              case Some => {
```
Or `_` is even better?